File: sword-means-offer/03/main.py
# -*- coding:utf-8 -*-
# @Time     : 2021/8/2 18:47
# @Author   : Ranshi
# @File     : main.py
from typing import List


class Solution:

    def findRepeatNumber(self, nums: List[int]) -> int:
        s = set()
        for x in nums:
            if x in s:
                return x
            else:
                s.add(x)
        return -1


if __name__ == '__main__':
    print(Solution().findRepeatNumber([2, 3, 1, 0, 2, 5, 3]))

File: sword-means-offer/03/main.go
package main

import "fmt"

// @Time     : 2021/09/25 23:15
// @Author   : Ranshi
// @File     : main.go

func findRepeatNumber(nums []int) int {
	set := make(map[int]bool, len(nums))
	for _, x := range nums {
		if value := set[x]; !value {
			set[x] = true
		} else {
			return x
		}
	}
	return -1
}

func main() {
	fmt.Printf("%v\n", findRepeatNumber([]int{2, 3, 1, 0, 2, 5, 3}))
}
